src/Dao/CategoryDao.php

```diff
@@ -10,40 +10,48 @@
 namespace OxidEsales\GraphQL\Example\Dao;
 
 use OxidEsales\EshopCommunity\Internal\Framework\Database\QueryBuilderFactoryInterface;
+use OxidEsales\GraphQL\Base\Service\LegacyServiceInterface;
 use OxidEsales\GraphQL\Example\DataObject\Category;
+use OxidEsales\GraphQL\Example\Exception\CategoryNotFoundException;
 
 class CategoryDao implements CategoryDaoInterface
 {
 
     /** @var QueryBuilderFactoryInterface $queryBuilderFactory */
     private $queryBuilderFactory;
 
+    /** @var LegacyServiceInterface */
+    private $legacyService;
+
     public function __construct(
-        QueryBuilderFactoryInterface $queryBuilderFactory
+        QueryBuilderFactoryInterface $queryBuilderFactory,
+        LegacyServiceInterface $legacyService
     ) {
         $this->queryBuilderFactory = $queryBuilderFactory;
+        $this->legacyService = $legacyService;
     }
 
-    public function getCategoryById(string $id, int $languageId, int $shopId): ?Category
+    /**
+     * @throws CategoryNotFoundException
+     */
+    public function getCategoryById(string $id, int $languageId): ?Category
     {
         $queryBuilder = $this->queryBuilderFactory->create();
         $title = $languageId === 0 ? "OXTITLE" : "OXTITLE_$languageId";
 
         $queryBuilder->select(['OXID', $title, 'OXPARENTID'])
                      ->from('oxcategories')
                      ->where($queryBuilder->expr()->andX(
-                         $queryBuilder->expr()->eq('OXID', ':oxid'),
-                         $queryBuilder->expr()->eq('OXSHOPID', ':shopid')
+                         $queryBuilder->expr()->eq('OXID', ':oxid')
                      ))
-                     ->setParameter('oxid', $id)
-                     ->setParameter('shopid', $shopId);
+                     ->setParameter('oxid', $id);
         $result = $queryBuilder->execute();
         if (!$result instanceof \Doctrine\DBAL\Driver\Statement) {
-            return null;
+            throw new CategoryNotFoundException("Error executing category query.");
```

Comment:
I think this is more expressive:
```suggestion
            throw new \UnexpectedValueException("Category-Query result must be instance of \Doctrine\DBAL\Driver\Statement");
```
To the client this will be a internal server error (and he should not know about this problem details anyway), but it will be logged in our log file.